src/symbolic_memory.ml

```diff
@@ -111,51 +133,81 @@ module M = struct
     let v0 = load_byte m a in
     loop a n 1 v0
 
+  (* TODO: *)
+  (* 1. Let pointers have symbolic offsets *)
+  (* 2. Let addresses have symbolic values *)
+  let calculate_address m (a : int32) : (Int32.t, Trap.t) Stdlib.Result.t =
+    match a.e with
+    | Val (Num (I32 i)) -> Ok i
+    | Ptr (base, offset) -> (
+      match Hashtbl.find_opt m.chunks base with
```

Comment:
You can use `Hashtbl.find` and match on `exception Not_found -> ... | size -> ...` to avoid an allocation in the most frequent case.